### PR TITLE
ds-z3-init and prepending ds env variables with `ACCELERATE_`

### PR DESCRIPTION
**Pull request by pacman100**
### What does this PR do?
1. Addresses #922

### CODE REVIEW

**Review by muellerzr — APPROVED** [review]
LG2M, I'll have the slow runners test this as well before we merge :)  (and review by @stas00 )

File: src/accelerate/commands/launch.py [hunk 1]
```diff
@@ -343,14 +343,14 @@ def launch_command_parser(subparsers=None):
     )
     deepspeed_args.add_argument(
         "--zero3_init_flag",
-        default=None,
+        default="true",
         type=str,
         help="Decides Whether (true|false) to enable `deepspeed.zero.Init` for constructing massive models. "
         "Only applicable with DeepSpeed ZeRO Stage-3.",
```
Comment (stas00) [inline src/accelerate/commands/launch.py:349]: may be add to `help` that `ACCELERATE_DEEPSPEED_ZERO3_INIT` env var can override it?

File: src/accelerate/commands/launch.py [hunk 2]
```diff
@@ -343,14 +343,14 @@ def launch_command_parser(subparsers=None):
...
     )
     deepspeed_args.add_argument(
         "--zero3_save_16bit_model",
-        default=None,
+        default="false",
         type=str,
         help="Decides Whether (true|false) to save 16-bit model weights when using ZeRO Stage-3. "
         "Only applicable with DeepSpeed ZeRO Stage-3.",
```
Comment (stas00) [inline src/accelerate/commands/launch.py:356]: may be add to `help` that `ACCELERATE_DEEPSPEED_ZERO3_SAVE_16BIT_MODEL` env var can override it?

**Review by stas00 — CHANGES_REQUESTED** [review]
Thank you very much for the improvements, @pacman100!

I left a few small suggestions.

File: src/accelerate/commands/launch.py (hunk 1 above)
Comment (pacman100) [reply src/accelerate/commands/launch.py:349]: Hello, `ACCELERATE_DEEPSPEED_ZERO3_INIT` is set using `zero3_init_flag` arg, so I don't think explicit mention is required as it is only used internally.

File: src/accelerate/commands/launch.py (hunk 2 above)
Comment (pacman100) [reply src/accelerate/commands/launch.py:356]: Same explanation as above

File: src/accelerate/commands/launch.py (hunk 1 above)
Comment (stas00) [reply src/accelerate/commands/launch.py:349]: ah, understood, thank you!

File: src/accelerate/utils/dataclasses.py [hunk 3]
```diff
@@ -377,24 +377,26 @@ def __post_init__(self):
...
 
             if self.offload_param_device is None:
-                self.offload_param_device = os.environ.get("DEEPSPEED_OFFLOAD_PARAM_DEVICE", "none")
+                self.offload_param_device = os.environ.get("ACCELERATE_DEEPSPEED_OFFLOAD_PARAM_DEVICE", "none")
 
             if self.zero3_save_16bit_model is None:
-                self.zero3_save_16bit_model = os.environ.get("DEEPSPEED_ZERO3_SAVE_16BIT_MODEL", "false") == "true"
+                self.zero3_save_16bit_model = (
+                    os.environ.get("ACCELERATE_DEEPSPEED_ZERO3_SAVE_16BIT_MODEL", "false") == "true"
+                )
```
[the remark this answers is not in the document]
Comment (pacman100) [reply src/accelerate/utils/dataclasses.py:399]: Hello, here dataclass object attributes are still `None` initially. In `__post_init__` call, the env variables which were set via [default] cmd args are used for setting dataclass object attributes.

File: src/accelerate/utils/dataclasses.py [hunk 4]
```diff
@@ -417,7 +419,7 @@ def __post_init__(self):
         self.deepspeed_config = self.hf_ds_config.config
         self.deepspeed_config["steps_per_print"] = float("inf")  # this will stop deepspeed from logging @ stdout
         if self.zero3_init_flag is None:
-            self.zero3_init_flag = os.environ.get("DEEPSPEED_ZERO3_INIT", "false") == "true"
+            self.zero3_init_flag = os.environ.get("ACCELERATE_DEEPSPEED_ZERO3_INIT", "false") == "true"
```
[the remark this answers is not in the document]
Comment (pacman100) [reply src/accelerate/utils/dataclasses.py:422]: Same explanation as above

File: src/accelerate/utils/dataclasses.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (stas00) [reply src/accelerate/utils/dataclasses.py:399]: my bad then! my apologies for the false alert - thank you for explaining, @pacman100!

**Review by stas00 — APPROVED** [review]

**Pull request merged**